specifyweb/frontend/js_src/lib/components/FormFields/Field.tsx

```diff
@@ -162,6 +162,11 @@ function Field({
          */
         updateValue(input.value, event.type === 'paste');
       }}
+      step={
+        validationAttributes.step === undefined
+          ? 'any'
+          : validationAttributes.step
+      }
```

Comment:
1. I'm not sure (or at least didn't know) that `any` is a valid value for the step attribute - I though if you want to allow any step, you just pass undefined to this attribute
2. Sorry to say, but this is not a fix, just a bad workaround. This change would ignore the `step` attribute completely, even if it was set by the user in the form definition. Instead, you have to investigate where the value for `validationAttributes.step` is coming from in the case when it's not desirable.

<details><summary>Hint</summary>
<p>

Search for references to `step` in this file - https://github.com/specify/specify7/blob/issue-4789/specifyweb/frontend/js_src/lib/utils/parser/definitions.ts

`step` should have a default value of `1` for integer fields. `step` must not have a default value for fields that accept floating point.

Then, update the tests to catch this error for the future

</p>
</details>

Reply:
this fix allows the user to enter any number in the number input if he didn't set a step in the xml. In the case he did set a step in the xml it will be respected

Reply:
I pushed a code with changes in definitions.ts too. The behavior stays the same

Reply:
but what I don't like with this version is that if you have no step defined and you change your input to be by example 9.3 you will have an alert Please choose a valid number, The nearest valid number are 9 and 10. Error we do not have with my first push. I'd be in favor of keeping the first code

Reply:
> This change would ignore the step attribute completely, even if it was set by the user in the form definition

Ah you are right - sorry, I was too quick to jump on this

> I'm not sure (or at least didn't know) that any is a valid value for the step attribute

you are correct - `any` is a valid option for this attribute. And `1` is the implicit default value. I didn't know that

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step#:~:text=If%20not%20explicitly%20included%2C%20step%20defaults%20to%201%20for%20number%20and%20range%2C%20and%201%20unit%20type%20(minute%2C%20week%2C%20month%2C%20day)%20for%20the%20date/time%20input%20types.%20The%20value%20must%20be%20a%20positive%20number%20%2D%20integer%20or%20float%20%E2%80%94%20or%20the%20special%20value%20any